### ISSUE TITLE
There's an issue with capabilities where only full site admins can use the new editor. We'll need to also allow users with Event Manager roles to use the new editor (and other roles with the required event editing capabilities).

### ISSUE THREAD

**Issue by joshfeck** [eventespresso/event-espresso-core#1830]
This may be a REST API issue where it's been set to require manage_options capability.

**Comment by mnelson4** [comment]
> This may be a REST API issue where it's been set to require manage_options capability.

Yeah it's a deficiency of the EE REST API. The capabilities related to writing data were tricky and so not in the first version of write endpoints. We've never gotten around to adding them. I think there are three options:

1. So we could fully implement capabilities in the REST API -ie, you can only edit events if you have that capability etc. iirc last time Darren and I discussed it, we thought it might require adding some new capabilities. It's a fairly tricky system to implement, and will probably take a few weeks. Lastly, it will introduce a few weird situations for the UI (eg when users have the capabilities needed to edit events but not datetimes).

2. An alternative quickfix would be to change the required capability from `manage_options` to one event managers have, like maybe `ee_read_ee` or `ee_edit_private_events`. Or we could introduce a new one like `ee_edit_via_rest_api` and use that. This would obviously not provide the exact fine-grained capabilities everyone wants, and there's some potential for abuse (eg, if one event manager has capability to edit their OWN events, but not others', and they write some script to use the REST API, they could use this `ee_edit_via_rest_api` capability to edit events they're not supposed to.) 

So that would be easy, but theoretically there could be problems.

3. I saw @manzoorwanijk 's made a bunch of progress on EE GraphQL API. Maybe it would be better building this feature into that, rather than the REST API.
